pkg/util/certs/certs.go

```diff
@@ -0,0 +1,55 @@
+package certs
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"net/http"
+)
+
+// makeTLSConfig returns a TLS configuration with a CA
+// pool containing certificates from certs.
+func makeTLSConfig(certs []*x509.Certificate) *tls.Config {
+	caPool := x509.NewCertPool()
+	for _, cert := range certs {
+		caPool.AddCert(cert)
+	}
+	return &tls.Config{RootCAs: caPool}
+}
+
+// MakeTLSTransport returns an HTTP transport with a TLS
+// configuration containing CA certificates from certs.
+func MakeTLSTransport(certs []*x509.Certificate) *http.Transport {
+	cfg := makeTLSConfig(certs)
+	return &http.Transport{TLSClientConfig: cfg}
+}
+
+// CertsFromPEM parses pemCerts into a list of certificates.
+func CertsFromPEM(pemCerts []byte) ([]*x509.Certificate, error) {
```

Comment:
It's really unfortunate that we need to reproduce this code instead of being able to use the stdlib's [`CertPool.AppendCertsFromPEM`][1].  I guess AWS doesn't take a CertPool for wherever this is being consumed?  Can we bump Go's [`DefaultTransport`][2] `TLSClientConfig` [`RootCAs`][3] instead of talking to the AWS libraries directly?

[1]: https://golang.org/pkg/crypto/x509/#CertPool.AppendCertsFromPEM
[2]: https://golang.org/pkg/net/http/#RoundTripper
[3]: https://golang.org/pkg/crypto/tls/#Config

Reply:
Oh, actually, you can drop `CertsFromPEM` and use `AppendCertsFromPEM` instead in `ensureDNSClientsTLSTransport` to construct a transport.  Then pass that transport in instead of your current 

```golang
	m.route53.Client.Config.HTTPClient.Transport = certsutil.MakeTLSTransport(certs)
	m.elb.Client.Config.HTTPClient.Transport = certsutil.MakeTLSTransport(certs)
	m.tags.Client.Config.HTTPClient.Transport = certsutil.MakeTLSTransport(certs)
```

which makes three identical transports.

Reply:
Annoying right? The cloud provider libs are idiosyncratic... Azure won't use the stdlib transport the way you would want either, so we have to wire up everything in a way the lib demands